### DISCUSSION TITLE
Core uses

### DISCUSSION THREAD

**Discussion by dpkhatua** [epochpic/epoch#568] [General]
Hi Everyone,
I am doing a simulation using 50 cores, but when I checked the core usage, I found that only 23 cores are being used by the simulation. I guess this making the calculation bit slower. Can anyone suggest what changes should I make to use all the cores? 

Thanks.

**Comment by Status-Mirror** [comment]
Hey @Khatua03 

Can you post the input deck?

EPOCH parallelises by domain decomposition - each core gets to work on a separate part of the simulation window. There is a minimum cell range per core, so maybe your simulation window is too small?

Cheers,
Stuart

**Reply by dpkhatua** [reply]
Hi @Status-Mirror ,
Thanks for the quick reply. Yes, indeed, my simulation window is small. Please take a look at the input file.
[input.txt](https://github.com/Warwick-Plasma/epoch/files/13125851/input.txt)

Thanks.

**Reply by Status-Mirror** [reply]
The important feature is cell count - here you have a (1000 x 4000) domain, which should be big enough to split between 50 cores.

When EPOCH is running, it should tell you the domain decomposition (number of processing elements in each direction). Something like:

```
 Code is running on 8 processing elements
```

or

```
 Initial conditions complete and valid.
 Calculating optimal processor topology
 Processor subdivision is            1           8
```

Do these numbers imply 23 cores instead of 50?

P.S. I don't have access to a 50 core machine for debugging, so I won't be able to verify this myself.

**Reply by dpkhatua** [reply]
Thanks Stuart. At the beginning, it starts with 50 cores (the processor subdivision is 1 50). But eventually, the number of cores used for the simulation decreases with time.

Thanks.
Durga Prasad.
